src/entity/Entity.php

```diff
@@ -658,6 +658,18 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 			$ev = new EntityDamageEvent($this, EntityDamageEvent::CAUSE_VOID, 10);
 			$this->attack($ev);
 			$hasUpdate = true;
+		}else{
+			$entityBlock = $this->getWorld()->getBlock($this->getLocation());
+			$entityBox = $this->getBoundingBox()->expandedCopy(0.001, 0.1, 0.001);
+			foreach (Facing::ALL as $face) {
+				$block = $entityBlock->getSide($face);
+				foreach ($block->getCollisionBoxes() as $blockBox) {
+					if ($entityBox->intersectsWith($blockBox)) {
+						$block->onEntityCollide($this, Facing::opposite($face));
+						break;
```

Comment:
I'm not convinced we should break here. Entities can collide with multiple blocks at once.

Reply:
If player collide with an AABB of block, we don't need to check all AABB of this block so we break. 
All faces are checked with the foreach above

Reply:
That may be the case right now as only magma uses this. But in the future more blocks might use it with different behaviours.

Reply:
For exemple, if we take cauldron that return multiple AABB in `getCollisionBoxes()`. If player collide with two AABB for the same face, we have two `onEntityCollide` call for the same face. It's illogic

Reply:
@dktapps He's breaking `foreach ($block->getCollisionBoxes() as $blockBox) {`, in case a collision with that block has already been found, `foreach (Facing::ALL as $face) {` will continue iterating.

Reply:
Right. In that case it should be using this: https://github.com/pmmp/PocketMine-MP/blob/9cab72ed5907b3d26040e88f89b7353b0de03ec3/src/block/Block.php#L841